src/scheduled_tasks.rs

```diff
@@ -450,6 +461,64 @@ async fn update_banned_when_expired(pool: &mut DbPool<'_>) {
   }
 }
 
+/// Find all unpublished posts with scheduled date in the future, and publish them.
+async fn publish_scheduled_posts(context: &Data<LemmyContext>) {
+  let pool = &mut context.pool();
+  let conn = get_conn(pool).await;
+
+  match conn {
+    Ok(mut conn) => {
+      let scheduled_posts: Vec<_> = post::table
+        .inner_join(community::table)
+        .inner_join(person::table)
+        .filter(post::scheduled_publish_time.is_not_null())
+        .filter(not(post::deleted))
+        .filter(not(post::removed))
+        .filter(not(person::banned))
+        .filter(not(person::deleted))
+        .get_results::<(Post, Community, Person)>(&mut conn)
+        .await
+        .inspect_err(|e| error!("Failed to read unpublished posts: {e}"))
+        .ok()
+        .unwrap_or_default();
+
+      for (post, community, person) in scheduled_posts {
+        if post.scheduled_publish_time < Some(Utc::now()) {
+          // scheduled time is not reached yet, ignore
+          continue;
+        }
+        let is_banned = check_community_ban(&person, community.id, &mut context.pool())
+          .await
+          .is_err();
+        if is_banned {
+          // user is banned, ignore the scheduled post
+          continue;
+        }
+
+        // mark post as published in db
+        diesel::update(post::table)
+          .filter(post::id.eq(post.id))
+          .set(post::scheduled_publish_time.eq(None::<DateTime<Utc>>))
+          .execute(&mut conn)
+          .await
+          .inspect_err(|e| error!("Failed update scheduled post: {e}"))
+          .ok();
```

Comment:
Should we set `post.published` to the current time as well?

Reply:
Hrm... up to you. In that case maybe the `post/create` should do that as well. Also I'd probably add a SQL index on the column, because checking for `post.published = null` vs `post.published > current_date` could be slower.